subprojects/gst-plugins-bad/gst-libs/gst/d3d11/gstd3d11memory.cpp

```diff
@@ -1224,6 +1227,65 @@ gst_d3d11_memory_get_processor_output_view (GstD3D11Memory * mem,
   return mem->priv->processor_output_view;
 }
 
+/**
+ * gst_d3d11_memory_get_d2d1_render_target:
+ * @mem: a #GstD3D11Memory
+ * @factory: a #ID2D1Factory
+ *
+ * Returns: (transfer none) (nullable): a pointer to the
+ * ID2D1RenderTarget or %NULL if it can't be created for this type of memory
+ *
+ * Since: Added by vicon
+ */
+ID2D1RenderTarget* gst_d3d11_memory_get_d2d1_render_target (GstD3D11Memory* mem,
+    ID2D1Factory* factory)
+{
+    g_return_val_if_fail (gst_is_d3d11_memory(GST_MEMORY_CAST(mem)), NULL);
+    g_return_val_if_fail (mem->priv != NULL, NULL);
+    g_return_val_if_fail (factory != NULL, NULL);
+
+    if (mem->priv->d2d1_render_target != NULL)
+        return mem->priv->d2d1_render_target;
+
+    const guint bind_flags = D3D11_BIND_SHADER_RESOURCE | D3D11_BIND_RENDER_TARGET;
+    if ((mem->priv->desc.BindFlags & bind_flags) != bind_flags) {
+        GST_ERROR ("Texture bind flags (%u) are not compatible", mem->priv->desc.BindFlags);
+        return NULL;
+    }
+
+    D2D1_RENDER_TARGET_PROPERTIES props =
+        D2D1::RenderTargetProperties(
+            D2D1_RENDER_TARGET_TYPE_DEFAULT,
+            D2D1::PixelFormat(DXGI_FORMAT_UNKNOWN, D2D1_ALPHA_MODE_PREMULTIPLIED),
+            96,
+            96);
```

Comment:
Why 96? Isn't better to set it to 0.0 which will make `RenderTargetProperties` use default dpiX and dpiY?